plugin-template

```diff
@@ -0,0 +1,269 @@
+        necessary_dir_structure = os.path.dirname(os.path.join(plugin_root_dir, destination_relative_path))
+
+        if not os.path.exists(necessary_dir_structure):
+            os.makedirs(necessary_dir_structure)
+
+        if relative_path.endswith('.j2'):
+            template = env.get_template(relative_path)
+            destination = destination_relative_path[:-len('.j2')]
+            write_template_to_file(template, plugin_root_dir, destination, config)
+            files_templated += 1
+            if verbose:
+                print(f"Templated file: {relative_path}")
+        else:
+            shutil.copyfile(
+                os.path.join(section_template_dir, relative_path),
+                os.path.join(plugin_root_dir, relative_path)
+            )
+            files_copied += 1
+            if verbose:
+                print(f"Copied file: {relative_path}")
+
+    print(f"Section: {name} \n    templated: {files_templated}\n    copied: {files_copied}")
+    return 0
+
+
+def generate_relative_path_set(root_dir):
+    """
+    Create a set of relative paths within the specified directory.
+    """
+    applicable_paths = set()
+    for root, dirs, files in os.walk(root_dir, topdown=False):
+        for file_name in files:
+            template_abs_path = os.path.join(root, file_name)
+            template_relative_path = os.path.relpath(template_abs_path,
+                                                     root_dir)
+            applicable_paths.add(template_relative_path)
+    return applicable_paths
+
+
+def write_template_to_file(template, plugin_root_dir, relative_path, config):
+    """
+    Render template with values from the config and write it to the target plugin directory.
+    """
+
+    with tempfile.NamedTemporaryFile(mode='w', dir=plugin_root_dir, delete=False) as fd_out:
+        tempfile_path = fd_out.name
+        fd_out.write(template.render(**config))
+        fd_out.write('\n')
+
+        destination_path = os.path.normpath(os.path.join(plugin_root_dir, relative_path))
+        os.rename(tempfile_path, destination_path)
+
+        if destination_path.endswith('.sh'):
+            mode = stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IRGRP | stat.S_IWGRP | \
+                stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH
+        else:
+            mode = stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP | stat.S_IROTH
+        os.chmod(destination_path, mode)
+
+
+def generate_config(plugin_name, plugin_app_label):
+    """
+    Generates a default config for a plugin
+
+    Args:
+        plugin_name (str): the full name of the plugin repository name (e.g. pulp_file)
+        plugin_app_label (str): the Django app label for the plugin - usually the part after
+            pulp_ or pulp-
+
+    Returns:
+        config (dict): a dictionary containing a default template configuration for the plugin
+    """
+    config = {}
+    config['plugin_name'] = plugin_app_label
```

Comment:
This looks odd. I would have expected:
```python
config['plugin_name'] = plugin_name
config['plugin_app_label'] = plugin_app_label
```

Reply:
This is due to the names that were used when the plugin template was first created. I'll try to change this to be more straight forward.